src/software/ai/hl/stp/stp.cpp

```diff
@@ -21,8 +22,15 @@
 STP::STP(std::function<std::unique_ptr<Play>()> default_play_constructor,
          std::shared_ptr<const AIControlConfig> control_config, long random_seed)
     : default_play_constructor(default_play_constructor),
+      current_play(nullptr),
+      readable_robot_tactic_assignment(),
       random_number_generator(random_seed),
-      control_config(control_config)
+      control_config(control_config),
+      override_play_name(""),
+      previous_override_play_name(""),
+      override_play(false),
+      previous_override_play(false),
+      current_game_state()
```

Comment:
Might be worth adding some sort of abstraction for all the overriding stuff, seems like this class is getting a bit bloated.

Reply:
this is nothing new, just setting default values, so will keep it in mind but not in this pr

Reply:
tbh it's not immediately obvious how to cleanly remember the previous play override